examples/transcription/websocket/callback_stream.rs

```diff
@@ -0,0 +1,53 @@
+use std::env;
+use std::time::Duration;
+
+use futures::stream::StreamExt;
+
+use deepgram::{
+    common::options::{Encoding, Endpointing, Language, Options},
+    Deepgram, DeepgramError,
+};
+
+static PATH_TO_FILE: &str = "examples/audio/bueller.wav";
+static AUDIO_CHUNK_SIZE: usize = 3174;
+static FRAME_DELAY: Duration = Duration::from_millis(16);
+
+#[tokio::main]
+async fn main() -> Result<(), DeepgramError> {
+    let deepgram_api_key =
+        env::var("DEEPGRAM_API_KEY").expect("DEEPGRAM_API_KEY environmental variable");
+
+    let dg_client = Deepgram::new(&deepgram_api_key)?;
+
+    let options = Options::builder()
+        .smart_format(true)
+        .language(Language::en_US)
+        .build();
+
+    let callback_url = env::var("DEEPGRAM_CALLBACK_URL")
+        .expect("DEEPGRAM_CALLBACK_URL environmental variable")
+        .parse()
+        .expect("DEEPGRAM_CALLBACK_URL not a valid URL");
+
+    let mut results = dg_client
+        .transcription()
+        .stream_request_with_options(options)
+        .keep_alive()
+        .encoding(Encoding::Linear16)
+        .sample_rate(44100)
+        .channels(2)
+        .endpointing(Endpointing::CustomDurationMs(300))
+        .interim_results(true)
+        .utterance_end_ms(1000)
+        .vad_events(true)
+        .no_delay(true)
+        .callback(callback_url)
```

Comment:
I think Cliff wanted to avoid calling it like this due to the different response format which would require to make everything optional.

Response for a callback is a single request ID as far as I know.

Worth checking if this is the case with streaming also

Code for handling callbacks like this for prerecorded which was reverted

https://github.com/deepgram-devs/deepgram-rust-sdk/pull/74/commits/868e2a72b4669aa63ef5caeaab1be4944015ef5e#diff-0b67a8008948b24f9efed7d0976098a96455c5647d48df027f54d739c7972a3eR18-R25

Example of how pre-recorded support the callback url

https://github.com/deepgram-devs/deepgram-rust-sdk/blob/main/examples/transcription/rest/callback.rs#L32